packages/kbn-investigation-shared/src/schema/investigation_item.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as t from 'io-ts';
+
+const esqlItemSchema = t.type({
+  title: t.string,
+  type: t.literal('esql'),
+  params: t.type({
+    esql: t.string,
+    suggestion: t.any,
+  }),
+});
+
+const investigationItemsSchema = esqlItemSchema; // replace with union with various item types
+
+const investigationItemSchema = t.intersection([
```

Comment:
@kdelemme Can you explain me this type? I am aware of using `intersection` when mixing required and optional props. Here both props are required `t.type`, I didn't see any `t.partial` here. Or is intersection used here to support the union with various item types later on?

Reply:
Certainly. intersection is just a way to concatenate types to make a new one. In this case, I'm using intersection with some common props for all items, and then the item schema which will become a union of different item at some point.